iterator.go

```diff
@@ -791,6 +791,10 @@ func (i *refCommitsIter) Advance() error {
 				return io.EOF
 			}
 
+			if err == plumbing.ErrObjectNotFound {
```

Comment:
lets reverse order of all these checks (why do we check EOF and Object not found if an error is nil?
```go
if err != nil {
if err == io.EOF {
 				i.refs = nil
  				return io.EOF
  			}
if err == plumbing.ErrObjectNotFound {
  continue
}
return err
```